config/runtime.exs

```diff
@@ -69,13 +69,22 @@ if config_env() in [:prod, :demo] do
       For example: yourdomain.example.com
       """
 
+  # Returns an IP address to bind the HTTP server, based on IPv6 support.
+  # This prevents startup failures on systems without IPv6 where the app would crash
+  # Falls back to IPv4 `{0, 0, 0, 0}` if IPv6 is not available.
+  ip =
+    case :inet.getaddr(~c"localhost", :inet6) do
+      {:ok, _addr} -> {0, 0, 0, 0, 0, 0, 0, 0}
+      {:error, _} -> {0, 0, 0, 0}
```

Comment:
Hey Eugen, I see that Erlang inet provides a bunch of specific [POSIX error codes](https://www.erlang.org/doc/apps/kernel/inet.html#module-posix-error-codes), maybe we could be a little more specific about the error we're looking for here, which I guess would be `eafnosupport`. Could you test this?

Reply:
After an eye-to-eye discussion we decided to disregard my suggestion and leave it as it is, because what happens with `:inet.getaddr` is also a domain resolution syscall, so there are a bunch of other errors that might happen and indicate the absence of ipv6 support. We'll continue investigating on more precise ipv6 detection approaches as a background research task.